.github/workflows/get-heroku-app-commit-index.yml

```diff
@@ -0,0 +1,69 @@
+name: Get Heroku slug commit and resource index
+
+on:
+  workflow_call:
+    inputs:
+      app_name:
+        description: Heroku app name
+        type: string
+        required: true
+    outputs:
+      commit_hash:
+        description: The commit hash of the latest Heroku slug for the app.
+        value: ${{ jobs.heroku_app_info.outputs.commit_hash }}
+      resource_index:
+        description: The RESOURCE_INDEX extracted from the env/production/config.json
+        value: ${{ jobs.heroku_app_info.outputs.resource_index }}
```

Comment:
> Or even better, perhaps, what if we exposed this deployment/version metadata via our own API endpoint?

I like this idea for the scheduled updates of the `RESOURCE_INDEX`. We'd still have to worry about the initial deployment of the server though. I'm leaning towards deploying the review apps ourselves (as you [suggested](https://github.com/nextstrain/nextstrain.org/pull/841#discussion_r1588367534)) rather than depending on the `config.json`.

Reply:
Or instead of provisioning the index in CI, can we provision the index during the [Heroku release phase](https://devcenter.heroku.com/articles/release-phase)? This would, for both production apps and review apps, solve the "which code to run, which index to use" issues, and I think overall makes the coupling pretty clear.

Reply:
> Or instead of provisioning the index in CI, can we provision the index during the [Heroku release phase](https://devcenter.heroku.com/articles/release-phase)?

(Hah, we've [come full circle](https://github.com/nextstrain/nextstrain.org/issues/829#issuecomment-2062219926) back to the Heroku release phase 😆)
I'm also fine with using the release phase if we are open to updating the server IAM user permissions.

Reply:
Oh boy, I'd missed/forgotten that!

Reply:
> This would, for both production apps and review apps, solve the "which code to run, which index to use" issues

The resource indexer is built under fail-fast design principles. For heroku review apps this is fine. For production apps less so. 

Reading through this PR the complexity really ramps up when we start dealing with review apps. I'd support removing that complexity from actions and changing the procfile to something like `web: create-local-index-if-review-app && node server.js ...`, but keeping the canary/production site updated via github actions as done in this PR (with suggested changes)

Reply:
> The resource indexer is built under fail-fast design principles. For heroku review apps this is fine. For production apps less so.

If the release phase errors, then the deploy is aborted. Nothing breaks, the site keeps working. We would have to address the release phase errors before deploying again.

> changing the procfile to something like `web: create-local-index-if-review-app && node server.js ...`,

But in that case we'd still need to grant permissions that the actual server doesn't need in order for review apps to read the inventories?

I think we can keep index generation in GitHub Actions for both production and review apps but simplify the workflows a lot by moving automatic deploy of review apps into CI. This then parallels the production process too, and brings them more in line.

Reply:
A lot of the complexity comes from keeping things in sync between GH Actions and Heroku, so to reduce complexity, we should just do everything in one or the other. Since we rely on GH Actions for CI (see https://github.com/nextstrain/nextstrain.org/issues/850), it makes sense to move everything into GH Actions.

Reply:
That sounds good to me. But let's talk about what that looks like for review apps?

Reply:
Yeah, I would like to scope this PR to just supporting separate indexes for prod/canary. The work on the review apps can be done separately.

Reply:
To keep things simple, I've opted for a separate bash script to grab the Heroku/deployment metadata. This ~fine for now since we not setting any `RESOURCE_INDEX` envvars in Heroku. 

I still really like the idea of surfacing the deployment metadata through our own API endpoint, so tracking that separately in https://github.com/nextstrain/nextstrain.org/issues/853